pl_bolts/models/rl/reinforce_model.py

```diff
@@ -115,122 +116,80 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
-            batch_actions.append(ep_actions)
-
-        # get qvals
-        batch_qvals = []
-        for reward in batch_rewards:
-            batch_qvals.append(self.calc_qvals(reward))
-
-        # flatten the batched outputs
-        batch_actions, batch_qvals, batch_rewards, batch_states = self.flatten_batch(
-            batch_actions, batch_qvals, batch_rewards, batch_states
-        )
+        assert isinstance(rewards[0], float)
 
-        return batch_qvals, batch_states, batch_actions, batch_rewards
-
-    @staticmethod
-    def flatten_batch(
-        batch_actions: List[List[Tensor]],
-        batch_qvals: List[List[Tensor]],
-        batch_rewards: List[List[Tensor]],
-        batch_states: List[List[Tuple[Tensor, Tensor]]],
-    ) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
-        """
-        Takes in the outputs of the processed batch and flattens the several episodes into a single tensor for each
-        batched output
-
-        Args:
-            batch_actions: actions taken in each batch episodes
-            batch_qvals: Q vals for each batch episode
-            batch_rewards: reward for each batch episode
-            batch_states: states for each batch episodes
-
-        Returns:
-            The input batched results flattend into a single tensor
-        """
-        # flatten all episode steps into a single list
-        batch_qvals = list(chain.from_iterable(batch_qvals))
-        batch_states = list(chain.from_iterable(batch_states))
-        batch_actions = list(chain.from_iterable(batch_actions))
-        batch_rewards = list(chain.from_iterable(batch_rewards))
+        cumul_reward = []
+        sum_r = 0.0
 
-        # stack steps into single tensor and remove extra dimension
-        batch_qvals = torch.stack(batch_qvals).squeeze()
-        batch_states = torch.stack(batch_states).squeeze()
-        batch_actions = torch.stack(batch_actions).squeeze()
-        batch_rewards = torch.stack(batch_rewards).squeeze()
+        for r in reversed(rewards):
+            sum_r *= self.gamma
+            sum_r += r
+            cumul_reward.append(sum_r)
 
-        return batch_actions, batch_qvals, batch_rewards, batch_states
+        return list(reversed(cumul_reward))
 
-    def loss(
+    def train_batch(
         self,
-        batch_qvals: List[Tensor],
-        batch_states: List[Tensor],
-        batch_actions: List[Tensor],
-    ) -> torch.Tensor:
+    ) -> Tuple[List[torch.Tensor], List[torch.Tensor], List[torch.Tensor]]:
         """
-        Calculates the mse loss using a batch of states, actions and Q values from several episodes. These have all
-        been flattend into a single tensor.
-
-        Args:
-            batch_qvals: current mini batch of q values
-            batch_actions: current batch of actions
-            batch_states: current batch of states
+        Contains the logic for generating a new batch of data to be passed to the DataLoader
 
         Returns:
```

Comment:
you should use the key work
```suggestion
        Yield:
```
https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#docstring-sections